dataland_analytics/analysis_scripts/scripts_with_api_admin/datarequests_by_filter.py

```diff
@@ -0,0 +1,156 @@
+from datetime import datetime
+
+import dataland_community
+import dataland_datasets
+import dataland_qa
+import pandas as pd
+
+from dataland_analytics.config import *
+
+"""Get all data requests matching specific filters. Possible filters can be found in the API description of Dataland's 
+community manager at https://dataland.com/community/swagger-ui/index.html#/request-controller/getDataRequests 
+"""
+
+
+def get_readable_string_from_datatype(datatype: str) -> str:
+    """
+    Returns a readable string from a datatype string; if input string is not found, the input string is returned.
+    """
+    data_type_string_dict = {
+        'eutaxonomy-financials': 'EU Taxonomy for Financials Companies',
+        'eutaxonomy-non-financials': 'EU Taxonomy for Non-Financial Companies',
+        'lksg': 'LkSG',
+        'p2p': 'WWF Pathways to Paris',
+        'sfdr': 'SFDR',
+        'vsme': 'VSME',
+        'esg-questionnaire': 'ESG Questionnaire for Corporate Schuldscheindarlehen',
+        'heimathafen': 'Heimathafen',
+        'additional-company-information': 'Additional Company Information',
+    }
+    # returns value if key 'datatype' is found, else returns 'datatype' as default
+    return data_type_string_dict.get(datatype, datatype)
+
+
+def find_matching_entries_by_filter(data, criteria):
+    """
+    Returns a list of data points from data which match the criteria.
+
+    Args:
+        data: A list of data, each of type dictionary
+        criteria: A list of metadata, each of type dictionary which keys have to be a subset of the keys of
+                            list_of_requests
+
+    Returns:
+        a sublist of data which match an entry in criteria
+    """
+    matches = []
+    for datum in data:
+        for criterion in criteria:
+            if all(datum.get(key) == value for key, value in criterion.items()):
+                matches.append(datum)
+                break
+
+    return matches
+
+
+def set_matching_entries_to_pending(requests, matching_criteria):
+    matches = []
+    for req in requests:
+        for criterion in matching_criteria:
+            if all(req.get(key) == value for key, value in criterion.items()):
+                req["Request Status"] = "Pending"
+                break
+
+    return matches
+
+
+dataset_api_configuration = dataland_datasets.Configuration(
+    host=DATALAND_SERVER + "api", access_token=PERSONAL_USER_API_KEY
+)
+
+community_api_configuration = dataland_community.Configuration(
+    host=DATALAND_SERVER + "community", access_token=PERSONAL_USER_API_KEY
+)
+
+qa_api_configuration = dataland_qa.Configuration(
+    host=DATALAND_SERVER + "qa", access_token=PERSONAL_USER_API_KEY,
+)
+
+# Define filter parameters
```

Comment:
Move this to the very top of the script as this variable should actually be changed by the user